src/corehost/corehost.cpp

```diff
@@ -89,10 +99,10 @@ int exe_start(const int argc, const pal::char_t* argv[])
     pal::string_t app_path;
     pal::string_t app_root;
     bool requires_v2_hostfxr_interface = false;
-
+	
 #if FEATURE_APPHOST
     pal::string_t embedded_app_name;
-    if (!is_exe_enabled_for_execution(&embedded_app_name))
+    if (!is_exe_enabled_for_execution_hack(&embedded_app_name))
```

Comment:
This is obviously broken... as is the apphost won't run in the existing (non-single-file) scenarios.